packages/mui-system/src/index.spec.tsx

```diff
@@ -57,3 +57,25 @@ function styleTransformTest() {
     }),
   });
 }
+
+function styleTypeTest() {
+  const gap = style({
+    prop: 'gap',
+    themeKey: 'spacing',
+  });
+
+  // theme should be optional
+  const CustomComponentWithNoTheme = styled('div')(compose(display, gap({ gap: 5 })));
```

Comment:
That's not how you would pass the `gap` function. The component styles implementation calls it with the props of the component. You don't call it, and then pass the results to the styling.

This only works because our style functions return `any`. But `compose` actually requires callables.

Try running this code in a runtime test. It'll fail because it accepts the second argument of `compose` to be a StyleFunction.

Reply:
@eps1lon I see, thanks for explaining in detail! I made the following [change](https://github.com/mui-org/material-ui/pull/28744/commits/52874a63c05d82cbed85fe7f8998f9204a38b9f8).